### ISSUE TITLE
Backups fails with LOGICAL_BACKUP_S3_REGION unbound variable

### ISSUE THREAD

**Issue by mdemri** [zalando/postgres-operator#826]
``` /dump.sh: line 43: LOGICAL_BACKUP_S3_REGION: unbound variable                                                                                                                                         │
│ pg_dumpall: error: pg_dump failed on database "druid", exiting                                                                                                                                        │
│                                                                   ```

**Comment by FxKu** [comment]
Unfortunately, posting just an error message oneliner does not give us enough information to help you. Please provide more details about your setup and try to find some stack traces related to this error in log files (Postgres, Patroni, Operator etc.)

**Comment by mdemri** [comment]
Sorry about that.

Operator version is 1.3.1
Postgres version is 12

**Comment by FxKu** [comment]
I was assuming you are using these versions. I need stack traces from logs :smiley:

**Comment by mdemri** [comment]
```│   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                       │
│                                  Dload  Upload   Total   Spent    Left  Speed                                                                                                                         │
│   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100  8045    0  8045    0     0   714k      0 --:--:-- --:--:-- --:--:--  714k                                          │
│   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                       │
│                                  Dload  Upload   Total   Spent    Left  Speed                                                                                                                         │
│   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   167  100   167    0     0  13916      0 --:--:-- --:--:-- --:--:-- 15181                                          │
│   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                       │
│                                  Dload  Upload   Total   Spent    Left  Speed                                                                                                                         │
│   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   167  100   167    0     0  15181      0 --:--:-- --:--:-- --:--:-- 15181                                          │
│   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                       │
│                                  Dload  Upload   Total   Spent    Left  Speed                                                                                                                         │
│   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100 11863    0 11863    0     0  1053k      0 --:--:-- --:--:-- --:--:-- 1053k                                          │
│ + dump                                                                                                                                                                                                │
│ + /usr/lib/postgresql/12/bin/pg_dumpall                                                                                                                                                               │
│ + compress                                                                                                                                                                                            │
│ + pigz                                                                                                                                                                                                │
│ ++ estimate_size                                                                                                                                                                                      │
│ ++ /usr/lib/postgresql/12/bin/psql -tqAc 'select sum(pg_database_size(datname)::numeric) from pg_database;'                                                                                           │
│ + aws_upload 6765584                                                                                                                                                                                  │
│ + declare -r EXPECTED_SIZE=6765584                                                                                                                                                                    │
│ ++ date +%s                                                                                                                                                                                           │
│ + PATH_TO_BACKUP=s3://*****/spilo/postgres-*****/75193a15-41bb-11ea-8c2f-02c3c18f0f6a/logical_backups/1581834642.sql.gz                                                     │
│ + args=()                                                                                                                                                                                             │
│ + [[ ! -z 6765584 ]]                                                                                                                                                                                  │
│ + args+=("--expected-size=$EXPECTED_SIZE")                                                                                                                                                            │
│ + [[ ! -z '' ]]                                                                                                                                                                                       │
│ /dump.sh: line 43: LOGICAL_BACKUP_S3_REGION: unbound variable    ```

**Comment by FxKu** [comment]
Now this logs helps me! The problem is that you are using the logical backup image provided by us, where we haven't reflected the new variable `LOGICAL_BACKUP_S3_REGION`, yet. Should be fixed now.

**Comment by mdemri** [comment]
Thanks! How can I stick the backup container version/tag to a working version? Now it's working with latest tag, so we're facing this kind of issues.

**Comment by mdemri** [comment]
@FxKu We're still getting this error.

**Comment by FxKu** [comment]
Ah my bad. The operator of v1.3.1 does not set this variable, therefore you are still getting this error. It seems you have used the logical backup image from the OS repo before which already had the `LOGICAL_BACKUP_S3_REGION`, am I right? Using the `latest` operator version should make this work. We are also planning another release later this month.

**Comment by peter-c-larsson** [comment]
Seeing this issue currently based on the tag and the latest image.

Any plans to update build the backup image here? (since I don't see the tags for 1.3.1 and 1.4.0
https://registry.opensource.zalan.do/v2/acid/logical-backup/tags/list

Running the 1.3.1 chart as well. Even if it is little confusing on this line:
https://github.com/zalando/postgres-operator/blob/v1.3.1/charts/postgres-operator/Chart.yaml#L3

Anyone know a working tag.

**Comment by FxKu** [comment]
@peter-c-larsson the highest master tag doesn't have region option: `master-57`

This should work with v1.3.1. I wonder, why `latest` gives you something different, though

**Comment by peter-c-larsson** [comment]
I think it would work too after some more investigation. The issue in my case was it was a bit random. What I did was add latest tag specifically and haven't tested it yet: registry.opensource.zalan.do/acid/logical-backup:latest
My assumption is that it failed if the job ran on a node where the faulty image existed previously according to default behaviour here:
https://kubernetes.io/docs/concepts/containers/images/#updating-images

**Comment by FxKu** [comment]
We have now merged it in our internal project, as `v1.4.0` is out for a while now.

The image is `registry.opensource.zalan.do/acid/logical-backup:master-58`.
`master-57` tag does not have the region option as mentioned above.
